### DISCUSSION TITLE
Add a "Rome: Restart LSP Server" command?

### DISCUSSION THREAD

**Discussion by lgarron** [rome/tools#3202] [Suggestions]
### Environment information

```bash
macOS 12.5.1
VSCode v1.71.0
Rome extension v0.14.1
```

### What happened?

While adopting Rome for an existing codebase, I regularly need to edit `rome.json` and work with the revisited changes. There is currently no way to restart the LSP from the command palette other than the very heavy-handed "Reload Window" command. This has several undesirable side effects, such as losing recent file edit (undo) history.

### Expected result

It would be nice to have a "Restart LSP Server" command, similar to other LSP extensions.

Automatically updating the LSP config when `rome.json` changes (#3195) would go a significant way towards many of the use cases for a "Restart LSP Server" command.

But there are some cases where it's useful to restart the LSP server even when `rome.json` hasn't changed, such as when you are re-compiling the LSP binary in-place (e.g. because you're recompiling it from scratch during developement/debugging of Rome itself).

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

**Comment by ematipico** [comment]
This is indeed something that we could have.

**Reply by MichaReiser** [reply]
Tracked in #3524
